src/Command/InfectionCommand.php

```diff
@@ -192,6 +194,11 @@ protected function execute(InputInterface $input, OutputInterface $output)
             return 1;
         }
 
+        // We only apply a memory limit if there isn't one set
+        if ($adapter instanceof PhpUnitAdapter && ini_get('memory_limit') === '-1') {
```

Comment:
What about other test frameworks, e.g. PhpSpec? shouldn't we care about it as well?

I'm sure if we even end up with PhpUnitAdapter for now, we should introduce an interface like `MemoryUsageAware` (or whatever you name it) and check it rather than hardcoding PHPUnit related stuff in the very base command

```php
if ($adapter instanceof MemoryUsageAware && ini_get('memory_limit') === '-1') {
    $this->applyMemoryLimitFromTestFrameworkProcess();
}


class PhpUnitAdapter implement MemoryUsageAware...
```

Reply:
- [x] MemoryUsageAware